gensim/corpora/dictionary.py

```diff
@@ -148,7 +148,7 @@ def doc2bow(self, document, allow_update=False, return_missing=False):
 
         token2id = self.token2id
         if allow_update or return_missing:
-            missing = sorted((x for x in iteritems(counter) if x[0] not in token2id), key=lambda x: (x[1], x[0]))
+            missing = sorted((x for x in iteritems(counter) if x[0] not in token2id), key=lambda x: x[0])
```

Comment:
`sort` works lexicographically by default; no need for the `key` parameter at all.